### PR TITLE
New trace flame graph

### PR DESCRIPTION
**Pull request by ccschmitz**
## Summary

Updates the flame graph so it correctly maps spans on their appropriate layers. After exploring a number of different third-party libraries, I decided to build this on my own using SVG.

Screenshot 2023-10-20 at 4 12 22 PM

Also cleans up the loading and empty states on the trace panel and removes an unnecessary import in the React component templates.

Here are some explorations with a couple third-party libraries:

* [flame-chart-js](https://stackblitz.com/edit/typescript-qwaeeb?file=index.ts)
* [react-flame-graph](https://stackblitz.com/edit/stackblitz-starters-ashwqh?file=src%2Findex.tsx)

Note that there are some follow ups coming to this PR:

1. Prevent any spans from overlapping by bumping them down to the next row if there is an overlapping span, similar to how Datadog handles this.
2. Update zoom controls to provide a better UI for controlling the zoom + pan of the graph, similar to what we provide in the session replay timeline.
3. We will consider rewriting using D3, which would provide some helpful APIs for manipulating visualization of the spans and transitioning elements, which we may want to handle better in the future.

## How did you test this change?

Local + PR preview click test. Note that performance is really bad in the Reflame previews for some reason, but things behave as expected in the [Render preview](https://frontend-pr-6891.onrender.com/).

## Are there any deployment considerations?

N/A - all client-side changes.

## Does this work require review from our design team?

Will have Julian click test and review in production!

### CODE REVIEW

File: frontend/package.json [hunk 1]
```diff
@@ -68,6 +70,7 @@
 		"rudder-sdk-js": "2.20.0",
 		"subscriptions-transport-ws": "^0.11.0",
 		"use-query-params": "^2.2.0",
+		"use-resize-observer": "^9.1.0",
```
Comment (ccschmitz) [inline frontend/package.json:73]: TODO: Remove all these packages

File: _templates/react/component/component.ejs.t [hunk 2]
```diff
@@ -3,7 +3,6 @@ to: frontend/src/<%= name %>.tsx
 ---
 
 import { Box, Text } from '@highlight-run/ui'
-import React from 'react'
```
Comment (ccschmitz) [inline _templates/react/component/component.ejs.t:6]: Nit, but these aren't needed anymore and figured I'd clean them up.

**Review by SpennyNDaJets — APPROVED** [review]
Makes sense to me! left a couple comments around the time durations, but overall is an improvement aligning the spans

Before:
Screenshot 2023-10-23 at 3 57 23 PM

After:
Screenshot 2023-10-23 at 3 58 31 PM

File: frontend/src/pages/Traces/TraceFlameGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+
+const ZOOM_SCALING_FACTOR = 100
+const MAX_TICKS = 6
+export const ticksHeight = 24
+export const outsidePadding = 4
+export const lineHeight = 18
+const defaultCanvasWidth = 660
+const timeUnits = [
+	{ unit: 'h', divider: 1000000000000000 },
+	{ unit: 'm', divider: 1000000000000 },
```
[the remark this answers is not in the document]
Comment (ccschmitz) [reply frontend/src/pages/Traces/TraceFlameGraph.tsx:23]: Yeah, I think that would make more sense - these large numbers are hard to read 😅

File: frontend/src/pages/Traces/TraceFlameGraph.tsx [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+	{ unit: 's', divider: 1000000000 },
+	{ unit: 'ms', divider: 1000000 },
+	{ unit: 'ns', divider: 1 },
```
[the remark this answers is not in the document]
Comment (ccschmitz) [reply frontend/src/pages/Traces/TraceFlameGraph.tsx:26]: Yeah, I think we may want to actually cap it at microseconds for now after thinking about this more. Thanks for mentioning it.

I'll also try to clean up the values on the ticks across the top timeline so we don't get all those decimal places.

**Comment by ccschmitz** [comment]
@SpennyNDaJets I'm going to be following up with a couple PRs on this shortly, so I think I'm going to merge this in and fast follow with the changes mentioned above since this is all still only visible for Highlight admins.

**Pull request merged**